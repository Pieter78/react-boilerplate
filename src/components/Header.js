import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { startLogout } from '../actions/auth';

export const Header = ({ clickLogout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" exact="true" to="/Dashboard">
          <h1>Boilerplate</h1>
        </Link>
        <button className="button button--link" onClick={clickLogout}>Logout</button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = dispatch => ({
  clickLogout: () => dispatch(startLogout()),
});

export default connect(undefined, mapDispatchToProps)(Header);
