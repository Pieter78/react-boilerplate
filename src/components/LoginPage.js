import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ clickLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Boilerplate</h1>
      <p>Tagline for app</p>
      <button className="button button--blue" onClick={clickLogin}>Login with Google</button>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => ({
  clickLogin: () => {
    dispatch(startLogin());
  },
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
