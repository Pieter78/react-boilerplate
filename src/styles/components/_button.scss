.button {
  border: none;
  color: white;
  padding: $s-size;
  display: inline-block;
  font-size: $font-size-large;
  line-height: 1;
  font-weight: 300;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--blue {
  background: $light-blue;
}

.button--secondary {
  background: #888888;
}