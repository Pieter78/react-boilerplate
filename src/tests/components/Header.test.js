import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

let clickLogout;

beforeEach(() => {
  clickLogout = jest.fn();
});

test('should render Header correctly', () => {
  const wrapper = shallow(<Header startLogout={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogout on button click', () => {
  const wrapper = shallow(<Header clickLogout={clickLogout} />);
  wrapper.find('button').simulate('click');
  expect(clickLogout).toHaveBeenCalled();
});


// should call startLogout on button click

// LoginPage - as above
