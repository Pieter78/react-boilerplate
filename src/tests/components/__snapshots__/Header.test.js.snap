// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        exact="true"
        replace={false}
        to="/Dashboard"
      >
        <h1>
          Boilerplate
        </h1>
      </Link>
      <button
        className="button button--link"
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
